master/buildbot/process/properties.py

```diff
@@ -80,6 +82,36 @@ def master(self):
     def master(self, value):
         self._master = value
 
+    @property
+    def sourcestamps(self):
+        if self.build is not None:
+            return self.build.getAllSourceStamps()
+        return self._sourcestamps
+
+    @sourcestamps.setter
+    def sourcestamps(self, value):
+        self._sourcestamps = value
+
+    @defer.inlineCallbacks
```

Comment:
So this means that `Properties.allChanges()` and `Properties.allFiles()` must raise an error when `build` is not set.

On the other hand, I hope that it is possible for an `IRenderable` to return a deferred. Otherwise, I am screwed if I want to access the data API from inside it.